File: backend-nestjs/auth-service/src/organization/tenant.entity.ts
import { Entity, Column, PrimaryColumn, CreateDateColumn } from 'typeorm';

@Entity('tenants')
export class TenantEntity {
  @PrimaryColumn({ type: 'bigint' })
  id: number;

  @Column({ length: 255 })
  name: string;

  @Column({ unique: true, length: 100 })
  slug: string;

  @Column({ name: 'plan_type', length: 50, default: 'free' })
  planType: string;

  @CreateDateColumn({ name: 'created_at' })
  createdAt: Date;
}

File: backend-java/auth-service/src/main/java/com/cdm/auth/entity/TenantEntity.java
package com.cdm.auth.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "tenants")
@Getter @Setter @NoArgsConstructor
public class TenantEntity extends BaseEntity {

    @Column(nullable = false, length = 255)
    private String name;

    @Column(nullable = false, unique = true, length = 100)
    private String slug;

    @Column(name = "plan_type", length = 50)
    private String planType = "free";
}
